x-pack/plugins/data_quality/server/helpers/get_unallowed_field_requests.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  MsearchMultisearchHeader,
+  MsearchMultisearchBody,
+} from '@elastic/elasticsearch/lib/api/types';
+import { AllowedValuesInputs } from '../schemas/get_unallowed_field_values';
+
+export const getMSearchRequestHeader = (indexName: string): MsearchMultisearchHeader => ({
+  expand_wildcards: ['open'],
+  index: indexName,
+});
+
+export const getMSearchRequestBody = ({
+  indexFieldName,
+  allowedValues,
+  from,
+  to,
+}: {
+  indexFieldName: string;
+  allowedValues: AllowedValuesInputs;
+  from: string;
+  to: string;
+}): MsearchMultisearchBody => ({
+  aggregations: {
+    unallowedValues: {
```

Comment:
consider replacing

```ts
unallowedValues: {
```

with

```ts
[indexFieldName]: {
```

because the latter will echo the requested field in the response